xbmc/input/remote/IRemoteControl.h

```diff
@@ -0,0 +1,93 @@
+/*
+ *      Copyright (C) 2018 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+/*!
+  * \brief Interface for remote control
+  */
+class IRemoteControl
+{
+public:
+  virtual ~IRemoteControl() = default;
+
+  /*!
+   * \brief Get IR mappings file name
+   */
+  virtual std::string GetMapFile() = 0;
+
+  /*!
+   * \brief Call once during input manager initialization
+   */
+  virtual void Initialize() = 0;
+
+  /*!
+   * \brief Deinitialize remote control
+   */
+  virtual void Disconnect() = 0;
+
+  /*!
+   * \brief Get the button readed from IR
+   *
+   * \return A button code
+   */
+  virtual unsigned short GetButton() const = 0;
+
+  /*!
+   * \brief Get the time while a button was hold
+   *
+   * \return A hold time
```

Comment:
Plus 1 point for documenting the units, plus 2 points for including units in function name

e.g.
```c++
virtual unsigned int GetHoldTimeMs() const = 0;
```